openquakeplatform/openquakeplatform/static/js/irv_viewer.js

```diff
@@ -14,1043 +14,850 @@
+        for (var regionName in SVI) {
+            tempVal = (SVI[regionName] * sviWeight) + (RI[regionName] * riWeight);
+            IRI[regionName] = tempVal;
+        }
+    } else if (iriOperator == "Simple multiplication (ignore weights)") {
+        for (var regionName in SVI) {
+            tempVal = SVI[regionName] * RI[regionName];
+            IRI[regionName] = tempVal;
+        }
+    } else if (iriOperator == "Weighted multiplication") {
+        for (var regionName in SVI) {
+            tempVal = (SVI[regionName] * sviWeight) * (RI[regionName] * riWeight);
+            IRI[regionName] = tempVal;
+        }
+    }
 
-            scaledSviIndicator = jQuery.extend(true, {}, sviIndicator);
+    ///////////////
+    //// Scale ////
+    ///////////////
 
-            // Scale the svi values
-            var valueArray = [];
-            var scaleSVIvalues = [];
-            for (var v in scaledSviIndicator) {
-                valueArray.push(scaledSviIndicator[v]);
-            }
+    scale(IRI);
 
-            valueArray.shift();
-            var tempSVImin = Math.min.apply(null, valueArray),
-                tempSVImax = Math.max.apply(null, valueArray);
+    ////////////////////////////////////
+    //// Prep data for thematic map ////
+    ////////////////////////////////////
 
-            for (var j = 0; j < valueArray.length; j++) {
-                scaleSVIvalues.push( (valueArray[j] - tempSVImin) / (tempSVImax - tempSVImin) );
+    // Pass indicators into a 'newProperties' element
+    for (var ix = 0; ix < la.length; ix++) {
+        for (var key in IRI) {
+            if (key == la[ix].properties[selectedRegion]) {
+                la[ix].newProperties = {};
+                la[ix].newProperties['IRI'] = (IRI[key]).toFixed(5);
             }
-
-            var tempKeys = Object.keys(scaledSviIndicator);
-            tempKeys.shift();
-
-            for (var i = 0; i < tempKeys.length; i++) {
-                scaledSviIndicator[tempKeys[i]] = scaleSVIvalues[i];
+        }
+        for (var key in SVI) {
+            if (key == la[ix].properties[selectedRegion]) {
+                la[ix].newProperties['SVI'] = (SVI[key]).toFixed(5);
             }
-
-            scaledSviIndicator.plotElement = "svi"; // Lable within the object
-            delete scaledSviIndicator.SVIvalues;
-
-            //////////////////////////////////////////////
-            /////////// Create the PRI object ////////////
-            //////////////////////////////////////////////
-
-            for (var k in aalIndicator) {
-                delete aalIndicator[k];
+        }
+        for (var key in RI) {
+            if (key == la[ix].properties[selectedRegion]) {
+                la[ix].newProperties['IR'] = (RI[key]).toFixed(5);
             }
+        }
 
-            for (var k in scaledAalIndicator) {
-                delete scaledAalIndicator[k];
+        for (var key in riskIndicator[ix]) {
+            if (riskIndicator[ix] != 'region') {
```

Comment:
Same as above, this `if` wouldn't be necessary if riskIndicator was better structured